nvme.c

```diff
@@ -5214,14 +5214,15 @@ static int nvme_get_properties(int fd, void **pbar)
 	return err;
 }
 
-static void *mmap_registers(nvme_root_t r, struct nvme_dev *dev)
+static void *mmap_registers_prot(nvme_root_t r, struct nvme_dev *dev, int prot)
```

Comment:
I don't think we should pass in the flags for mmap. Just have a `bool writeable`. Also no need to rename the funtion.